CMakeModules/doxygen.cmake

```diff
@@ -0,0 +1,30 @@
+include(FindDoxygen)
+
+if(NOT DOXYGEN_FOUND)
+  message(STATUS "Doxygen not found.  Contract documentation will not be generated.")
+else()
+  message(STATUS "Doxygen found.  Contract documentation will be generated.")
+  set(DOXY_EOS_VERSION "${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH}" CACHE INTERNAL "Version string used in PROJECT_NUMBER.")
+  # CMake strips trailing path separators off of variables it knows are paths,
+  # so the trailing '/' Doxygen expects is embedded in the doxyfile.
+  set(DOXY_DOC_DEST_DIR "${CMAKE_BINARY_DIR}/docs" CACHE PATH "Path to the doxygen output")
+  set(DOXY_DOC_INPUT_ROOT_DIR "contracts" CACHE PATH "Path to the doxygen input")
+  if(DOXYGEN_DOT_FOUND)
+    set(DOXY_HAVE_DOT "YES" CACHE STRING "Doxygen to use dot for diagrams.")
+  else(DOXYGEN_DOT_FOUND)
+    set(DOXY_HAVE_DOT "NO" CACHE STRING "Doxygen to use dot for diagrams.")
+  endif(DOXYGEN_DOT_FOUND)
+  configure_file("eos.doxygen.in" "${CMAKE_BINARY_DIR}/eos.doxygen")
+  if(BUILD_DOXYGEN)
+     # !! Core developers who don't want to see the doxygen output on every build
```

Comment:
Please incorporate existing changes in the doxygen-related code in this branch.

Reply:
done